### PR TITLE
[risk=low][RW-4102] Indicate Workspace published state in UI (to FEATURED_WORKSPACE_ADMINs) 

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-about.tsx [hunk 1]
```diff
@@ -134,6 +134,12 @@ export const WorkspaceAbout = fp.flow(withUserProfile(), withCdrVersions())
     return workspace;
   }
 
+  // update the component state AND the store with the new workspace object
+  updateWorkspaceState(workspace: WorkspaceData): void {
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-about.tsx:138]: not entirely sure this is the right logic.  What I'd really like is for the state to update automatically when `currentWorkspaceStore` updates somehow

File: ui/src/app/pages/workspace/workspace-about.tsx [hunk 2]
```diff
@@ -164,27 +170,17 @@ export const WorkspaceAbout = fp.flow(withUserProfile(), withCdrVersions())
     }
   }
 
-  workspaceRuntimeBillingProjectId(): string {
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-about.tsx:167]: unrelated - this code is not used

File: ui/src/app/pages/workspace/workspace-about.tsx [hunk 3]
```diff
@@ -201,18 +197,20 @@ export const WorkspaceAbout = fp.flow(withUserProfile(), withCdrVersions())
   render() {
     const {profileState: {profile}, cdrVersionTiersResponse} = this.props;
     const {workspace, workspaceUserRoles, sharing, publishing} = this.state;
+    const published = workspace?.published;
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-about.tsx:200]: new syntax!

File: ui/src/app/pages/workspace/workspace-about.tsx (hunk 1 above)
Comment (calbach) [reply ui/src/app/pages/workspace/workspace-about.tsx:138]: Optional: I think a more natural way to do this is probably to use the `withCurrentWorkspace()` HOC. Reference workspace via props. Updating the workspace store should trigger the entire component to rerender.

This is a pretty old class component though, so I expect that refactor may uncover some issues.

**Review by calbach — APPROVED** [review]

**Pull request merged**